analyzer/codechecker_analyzer/analyzers/result_handler_base.py

```diff
@@ -92,8 +93,36 @@ def analyzer_action_str(self):
             os.path.join(self.buildaction.directory,
                          self.analyzed_source_file))
 
-        build_info = source_file + '_' + \
-            self.buildaction.original_command
+        # In case of "make 4.3" depending on compile-time options "make" tool
+        # can be built so a subprocess named cc1build will be logged by
+        # "CodeChecker log".
+        # See posix_spawn() option:
+        # https://lists.gnu.org/archive/html/info-gnu/2020-01/msg00004.html
+        # In this case the -o output argument of cc1build command is a randomly
+        # named temporary file. We can't afford dynamic parts in the original
+        # build command, because its hash is used for identification in the
+        # plist file name.
+        #
+        # TODO: This solution is considered a workaround, because the real
+        # question is why such a subprocess is logged? Can cc1build be always
+        # used as a traditional g++ command?
```

Comment:
I believe the connection between the two is the same as doing `clang` (the frontend) versus. `clang -cc1` (the compiler itself). There's definitely a _way_ to convert the compile command (that's what the driver does) but simply changing the "program name" is not going to work.

I do not have this `cc1build` binary. However, Clang "recently" introduced a compile option `CLANG_SPAWN_CC1` and a related flag which basically toggles Clang's behaviour between spawning a subprocess for the compiler itself (as *GCC* does with `cc1`/`cc1plus`), or doing the compilation in itself.

Have we ever encountered a similar issue with Clang, especially in the old days?

Reply:
No, I haven't encountered such an issue with Clang. However, I'll create an issue for checking what happens in case of `cc1build`, because we have some other use-cases where the compiler binary matters. For example can we gather implicit compiler info (include paths, default standard version, etc.) using this binary?
These things are reproducible under Archlinux using gcc and make tools from the standard Archlinux repo.